agent/util-scripts/pbench-tool-meister-stop

```diff
@@ -67,85 +78,80 @@ def main(argv):
-        logger.debug("tm_pids = %r", tm_pids)
-
-    ret_val = 0
-
-    logger.debug("terminating %d pids at localhost:%d", expected_pids, redis_port)
-    terminate_msg = dict(action="terminate", group=None, directory=None, args=None)
-    try:
-        num_present = redis_server.publish(
-            channel, json.dumps(terminate_msg, sort_keys=True)
-        )
-    except Exception:
-        logger.exception("Failed to publish terminate message")
-        ret_val = 1
-    else:
-        if num_present != expected_pids:
+        if sysinfo is None:
+            sysinfo = "default"
+        sts, item = verify_sysinfo(sysinfo)
+        if sts != 0:
+            logger.error('invalid sysinfo option, "{}"', item)
+            return sts
+
+    # The Redis server is always running on the local host with the CLI.
+    with Client(redis_host="localhost", redis_port=redis_port, logger=logger) as client:
+        # First we end the persistent tools
+        tool_dir = benchmark_run_dir / f"tools-{group}"
+        try:
+            tool_dir.mkdir(exist_ok=True)
+        except Exception as exc:
             logger.error(
-                "Failed to terminate %d pids, only encountered %d on the channel",
-                expected_pids,
-                num_present,
+                'failed to create tool output directory, "{}": {}', tool_dir, exc
             )
-            ret_val = 1
-
-    # The assumption/assertion here is that the tool meister "stop" command is
-    # run on the same node as the tool data sink. So we want to make sure
-    # anything "local" to this stop command is shut down gracefully before we
-    # report back to the user.  If tool meisters from remote nodes have
-    # already reported that they have received the "terminate" message, then
-    # we trust they will shutdown gracefully themselves.
-    if "ds" in tm_pids:
-        pid = tm_pids["ds"]["pid"]
-        logger.debug("waiting for tool-data-sink (%d) to exit", pid)
-        while is_running(pid):
-            time.sleep(0.1)
-    if "tm" in tm_pids:
-        for tm in tm_pids["tm"]:
-            if tm["hostname"] == full_hostname:
-                pid = tm["pid"]
-                logger.debug("waiting for local tool-meister (%d) to exit", pid)
-                while is_running(pid):
-                    time.sleep(0.1)
+        else:
+            client.publish(group, tool_dir, "end", None)
+        # Next we collect the system configuration information
+        if sysinfo != "none":
+            sysinfo_path = benchmark_run_dir / "sysinfo" / "end"
+            try:
+                sysinfo_path.mkdir(parents=True)
+            except Exception:
+                logger.error(
+                    "Unable to create sysinfo-dump directory base path: {}",
+                    sysinfo_path,
+                )
+            else:
+                logger.info("Collecting system information")
+                info_log("Collecting system information")
```

Comment:
This is hack to provide the same output as before.  We really should fix the log file handling on the agent side, but that would be for another commit.

Reply:
Tracked in https://github.com/distributed-system-analysis/pbench/issues/2155